### PR TITLE
Support for skid-steering and articulations.

### PR DESCRIPTION
**Pull request by michalpelka**
Support for skid-steering and articulations, fix limit handling with URDF import.

### CODE REVIEW

File: Gems/ROS2/Code/Source/VehicleDynamics/Utilities.cpp [hunk 1]
```diff
@@ -268,4 +243,31 @@ namespace ROS2::VehicleDynamics::Utilities
...
+    {
+        AZ::Transform hingeTransform{ AZ::Transform::Identity() };
+        if (data.m_isArticulation)
+        {
+            hingeTransform = AZ::Transform::Identity();
+        }
+        else
+        {
+            PhysX::JointRequestBus::EventResult(hingeTransform, AZ::EntityComponentIdPair(data.m_wheelEntity,data.m_wheelJoint), &PhysX::JointRequests::GetTransform);
+        }
```
Comment (lumberyard-employee-dm) [inline Gems/ROS2/Code/Source/VehicleDynamics/Utilities.cpp:270]: The if block isn't changing the hingeTransform value at all.
The condition can be converted to check if articulation is not enabled.
```suggestion
        if (!data.m_isArticulation)
        {
            PhysX::JointRequestBus::EventResult(hingeTransform, AZ::EntityComponentIdPair(data.m_wheelEntity,data.m_wheelJoint), &PhysX::JointRequests::GetTransform);
        }
```

**Review by lumberyard-employee-dm — APPROVED** [review]

**Comment by michalpelka** [comment]
Manually tested against (on Physx 4): 
- https://github.com/RobotecAI/o3de-ros2-gem-testing/blob/main/RobotControlTest/04_CONTROL_ACKERMANN_DRIVE/CONTROL_ACKERMANN_DRIVE_TESTCASE.md 
Proposed manual test on Physx 5
https://github.com/RobotecAI/o3de-ros2-gem-testing/pull/11

**Review by adamdbrw — APPROVED** [review]

File: Gems/ROS2/Code/Source/VehicleDynamics/Utilities.h [hunk 2]
```diff
@@ -67,4 +69,11 @@ namespace ROS2::VehicleDynamics::Utilities
     //! @returns ramped velocity according to time, acceleration and clamped to @param maxVelocity
     float ComputeRampVelocity(float targetVelocity, float lastVelocity, AZ::u64 deltaTimeNs, float acceleration, float maxVelocity);
 
+
+    void SetWheelRotationSpeed(const  VehicleDynamics::WheelDynamicsData& data, float wheelRotationSpeed);
```
Comment (jhanca-robotecai) [inline Gems/ROS2/Code/Source/VehicleDynamics/Utilities.h:73]: Please use `clang-format` to format the code
```suggestion
    void SetWheelRotationSpeed(const VehicleDynamics::WheelDynamicsData& data, float wheelRotationSpeed);
```

File: Gems/ROS2/Code/Source/VehicleDynamics/Utilities.cpp [hunk 3]
```diff
@@ -268,4 +243,27 @@ namespace ROS2::VehicleDynamics::Utilities
         }
         return AZStd::clamp(commandVelocity, -maxVelocity, maxVelocity);
     }
+
+    void SetWheelRotationSpeed(const  VehicleDynamics::WheelDynamicsData& data, float wheelRotationSpeed)
```
Comment (jhanca-robotecai) [inline Gems/ROS2/Code/Source/VehicleDynamics/Utilities.cpp:247]:
```suggestion
    void SetWheelRotationSpeed(const VehicleDynamics::WheelDynamicsData& data, float wheelRotationSpeed)
```

**Review by jhanca-robotecai — APPROVED** [review]
The code works correctly (tested with the proposed Physx 5 tests), please fix the formatting (including `setreg` file)

**Comment by michalpelka** [comment]
Will run clang-format on next PR on this code.

**Pull request merged**